ci(lint): golangci-lint-action cache reports stale '0 issues' on PRs that should fail

## Observed

PR #134 (docs-only spec changes) reported \`lint: pass\` in CI in **3.5s** while running on a base commit (\`d6b8494\`) that contained two known errcheck violations in \`cmd/gpu/setup.go\` (lines 80, 108 — unchecked \`sshClient.Close()\`).

Running the same lint locally on the same branch state produced:

\`\`\`
cmd/gpu/setup.go:80:18: Error return value of \`sshClient.Close\` is not checked (errcheck)
cmd/gpu/setup.go:108:24: Error return value of \`sshClient.Close\` is not checked (errcheck)
2 issues:
* errcheck: 2
\`\`\`

So CI was wrong. PR #135 had earlier flagged exactly these same two errors when it ran lint with a cold cache.

CI log excerpt for the false-pass run:

\`\`\`
Cache hit for: golangci-lint.cache-Linux-2938-aa0ad28fb3a54cf2d6376fc751fb80bd9f6b4e2b
Cache restored successfully
Running [.../golangci-lint run] in [...] ...
0 issues.
\`\`\`

The total \`golangci-lint run\` step was 3.5s — too fast for an actual analysis pass on this codebase (cold runs in PR #135 took 25s+).

After PR #135 merged the gpu fix into main and PR #134 was rebased, the lint correctly returned 0 issues — but again from a cache hit (~3s). It happens to be correct now because the fix is on main, but the **same cache poisoning pattern would mask any newly-introduced lint error** as long as a previous successful analysis cache exists for the build artifacts referenced.

## Root cause hypothesis

\`golangci/golangci-lint-action@v7\` caches the linter's internal analysis cache (\`~/.cache/golangci-lint/\`) under a key derived from \`go.mod\` / \`go.sum\` / config — not from source file content. When a previously-successful analysis result for a given Go package is in the cache, the linter trusts it on subsequent runs even if the source file content has materially changed in a way that would now violate a rule.

Combined with the action default \`only-new-issues: false\`, you'd expect a full re-analysis, but the cache short-circuits per-package work.

## Impact

- Silent lint regressions can slip into PRs without triggering a failing check.
- CI green is not a reliable signal that lint actually passed on the PR's tree.
- This is exactly the kind of failure that's invisible until you run lint locally.

## Suggested mitigations (pick one)

1. **Disable the action's cache** for lint runs:
   \`\`\`yaml
   - uses: golangci/golangci-lint-action@v7
     with:
       skip-cache: true
   \`\`\`
   Adds ~20s/run; correctness is worth it for a small repo.

2. **Pin a stricter cache invalidation interval** (currently default 7 days):
   \`\`\`yaml
   with:
     cache-invalidation-interval: 1
   \`\`\`
   Less safe but reduces stale-cache window.

3. **Run lint without the action wrapper** — install via \`go install\` and invoke directly so no per-package cache is reused across commits.

4. Pin to an older action version where this behavior is documented to differ — needs investigation.

## Reproduction

- Branch \`docs/spec-app-env-redesign\` at HEAD \`47cd66b\` (now merged as #134) — base commit \`d6b8494\` had unchecked \`sshClient.Close()\` errors but lint reported "0 issues" in CI.
- Locally: \`golangci-lint run ./...\` from that commit produces 2 errcheck failures.

## Notes

Discovered while reviewing PRs #134 + #135. PR #135's lint run with a cold cache flagged these errors correctly; PR #134's lint run with a warm cache did not. Both PRs are now merged and the symptom is no longer visible on \`main\`, but the underlying CI behavior persists.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -22,7 +22,15 @@ jobs:
       - uses: actions/setup-go@v5
         with:
           go-version: '1.26'
+      # skip-cache: the action's per-package analysis cache is keyed on
+      # go.mod/go.sum/config, not source content, so a previous "0 issues"
+      # result short-circuits re-analysis of files whose content would now
+      # violate a rule. See #136 for the concrete false-pass we saw on
+      # PR #134. ~20s extra per run is a fair trade for lint actually
+      # being a reliable signal.
       - uses: golangci/golangci-lint-action@v7
+        with:
+          skip-cache: true
 
   build:
     runs-on: ubuntu-latest

